functorch/csrc/DynamicLayer.cpp

```diff
@@ -435,7 +435,18 @@ void dynamicLayerFrontFallback(const c10::OperatorHandle& op, torch::jit::Stack*
   }
 #endif
   if (dynamicLayerStack.size() == 0) {
-    sanityCheckStack(op, stack);
+    // total hack: for now, the logic I added to generate the vmap rule
+    // doesn't play well with DynamicLayer (only one layer of vmap works right now).
+    // Why? In the generated batching rule, I effectively want to treat it as a "composite kernel",
+    // and have it run the to the python-defined forward function. But:
+    // (1) I want to go there through the dispatcher so other functionalities can run (e.g. AMP).
```

Comment:
Actually, maybe I should just be treating this the same way that DynamicLayer already treats composite ops - just directly call into the composite function. That means that stuff like AMP will run on the base ops and not the composite ops, but maybe that's the right behavior (unless the user wants to write a custom "AMP rule" for their op)

Reply:
(ended up doing this, although there's another issue with disabling autograd that I left a comment about)